conversation_classification/prediction_utils/user_features.py

```diff
@@ -0,0 +1,238 @@
+import json
+import pickle as cPickle
+import numpy as np
+from scipy.sparse import csr_matrix
+import random
+import matplotlib.pyplot as plt
+from collections import defaultdict
+import math
+import re
+import copy
+
+def attacker_profile(document, user_infos, ASPECTS):
+    """
+      Given the user information in conversations and a set of aspects to be inspected
```

Comment:
A little more information about the return data structure would be useful. For example, how are "No Gap", "New Comer", etc returned?

Reply:
Information added.